addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1954,7 +1954,10 @@ export class OdooEditor extends EventTarget {
             }
         }
         let insertedZws;
-        if (sel && !sel.isCollapsed && !range.startOffset && !range.startContainer.previousSibling) {
+        let { startContainer: start, startOffset, endContainer: end, endOffset } = range;
+        const [startBlock, endBlock] = [closestBlock(start), closestBlock(end)];
+        const [firstLeafOfStartBlock, lastLeafOfEndBlock] = [firstLeaf(startBlock), lastLeaf(endBlock)];
+        if (sel && !sel.isCollapsed && !startOffset && !start.previousSibling) {
```

Comment:
Did this need to change or is it a refactoring?

Reply:
I destructured the range above so i changed it, but as you said I'll do as little change as possible.